cli/src/commands/branch.rs

```diff
@@ -349,6 +349,21 @@ fn cmd_branch_rename(
         return Err(user_error(format!("Branch already exists: {new_branch}")));
     }
 
+    if view
+        .remote_branches_matching(
+            &StringPattern::exact(old_branch),
+            &StringPattern::everything(),
+        )
+        .count()
+        > 0
```

Comment:
This can be written `.next().is_some()`, which will terminate early instead of exhausting the iterator, but that shouldn't make a difference in practice in this case. Just a good pattern to keep in mind.

Reply:
Thanks. Noted the pattern. And Clippy helpfully suggests I can replace .filter().next() with .find()

Reply:
...aaaand Clippy suggests any() which takes a predicate anyway.